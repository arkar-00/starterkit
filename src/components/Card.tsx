import {StyleSheet, Text, View} from 'react-native';
import React from 'react';

const Card = ({title, content}: {title: string; content: string}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.content}>{content}</Text>
    </View>
  );
};

export default Card;

const styles = StyleSheet.create({
  container: {
    padding: 16,
    backgroundColor: '#fff',
    borderRadius: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 8,
  },
  content: {
    fontSize: 14,
    color: '#333',
  },
});
