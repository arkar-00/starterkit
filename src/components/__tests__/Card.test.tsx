import {render} from '@testing-library/react-native';
import Card from '../Card';

describe('Card', () => {
  const mockTitle = 'Test Title';
  const mockContent = 'This is a test content for the card component.';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders Card with title and content', () => {
    const {getByText} = render(<Card title={mockTitle} content={mockContent} />);

    expect(getByText(mockTitle)).toBeTruthy();
    expect(getByText(mockContent)).toBeTruthy();
  });

  test('should render correctly and match snapshot', () => {
    // Render the Card component with sample title and content
    const {toJSON} = render(
      <Card title="Test Title" content="This is the test content for the card." />,
    );

    // Get the JSON tree from the rendered component
    const tree = toJSON();

    // Expect the component's rendered tree to match the most recent snapshot.
    // If no snapshot exists, one will be created. If it differs, the test fails.
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with different content and match snapshot', () => {
    const {toJSON} = render(
      <Card
        title="Another Card Example"
        content="This card contains different content to ensure the snapshot testing captures variations. It might be longer."
      />,
    );
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
  });
});
