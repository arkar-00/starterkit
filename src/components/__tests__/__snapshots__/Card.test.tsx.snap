// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly and match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 5,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Test Title
  </Text>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 14,
      }
    }
  >
    This is the test content for the card.
  </Text>
</View>
`;

exports[`Card should render correctly with different content and match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 5,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Another Card Example
  </Text>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 14,
      }
    }
  >
    This card contains different content to ensure the snapshot testing captures variations. It might be longer.
  </Text>
</View>
`;
